### PR TITLE
Partitioner: workflow to add an MD array

### PR DESCRIPTION
**Pull request by ancorgs**
This allows to create a MD device, defining its level (aka RAID type), name and the list of devices. It also can be formatted, encrypted, etc.

https://trello.com/c/YDRIZHHh/700-8-storageng-expert-partitioner-create-md-array

### CODE REVIEW

File: src/lib/y2partitioner/sequences/md_controller.rb [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      #
+      # It removes any previous children (like filesystems) from the devices and
+      # adapts the partition id if possible.
+      #
+      # @param devices [Array<Y2Storage::BlkDevice>]
+      def devices_in_md=(devices)
+        md.devices.each { |dev| md.remove_device(dev) }
+        devices.each do |dev|
+          dev.adapted_id = Y2Storage::PartitionId::RAID if dev.is?(:partition)
+          dev = dev.encryption if dev.encrypted?
```
Comment (joseivanlopez) [inline src/lib/y2partitioner/sequences/md_controller.rb]: @aschnell  Does it make sense to reuse the encryption when adding a partition to a MD?

Comment (ancorgs) [reply src/lib/y2partitioner/sequences/md_controller.rb]: In other words, if a partition is already encrypted and the user chooses to add that partition to the RAID, should we add the encrypted device to the RAID or should we delete the encryption device and add only the plain partition?

File: src/lib/y2partitioner/widgets/format_and_mount.rb [hunk 2]
```diff
@@ -667,8 +672,53 @@ def items
...
+      end
+
+      # @macro seeAbstractWidget
+      def enable
+        @selector.enable if @selector
+      end
+
+      # @macro seeAbstractWidget
+      def disable
+        @selector.enable if @selector
```
Comment (joseivanlopez) [inline src/lib/y2partitioner/widgets/format_and_mount.rb]: Should it be `@selector.disable` ?

Comment (ancorgs) [reply src/lib/y2partitioner/widgets/format_and_mount.rb]: Indeed. Checked and I introduced a small bug. Fixing

File: src/lib/y2partitioner/sequences/md_controller.rb (hunk 1 above)
Comment (aschnell) [reply src/lib/y2partitioner/sequences/md_controller.rb]: In theory it should work to use encrypted devices for MD RAID. But I think that was so far not supported and may not work in practice.

File: src/lib/y2storage/callbacks/commit.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env ruby
```
Comment (joseivanlopez) [inline src/lib/y2storage/callbacks/commit.rb]: Shebang necessary?

**Pull request merged**